src/java.base/share/classes/jdk/internal/classfile/impl/AbstractPoolEntry.java

```diff
@@ -446,7 +446,7 @@ void writeTo(BufWriterImpl pool) {
                             throw new IllegalArgumentException();
                         }
                         int byteLengthFinal = byteLength;
-                        pool.patchInt(pool.size() - i - 2, 2, byteLengthFinal);
+                        pool.patchU2(pool.size() - i - 2, byteLengthFinal);
```

Comment:
This change is causing conflict with your utf8 writing enhancement. Can you revert this change and squash commits, or merge latest master to fix this conflict?